scripts/release/generate-changelog.js

```diff
@@ -73,53 +87,51 @@ function getChangesList() {
   const lastTagName = command`git describe --tags ${lastTagHash}`.run()
 
   const commits = command`git log ${lastTagName.trimEnd()}..HEAD --pretty=format:%s`.run()
-
   const changesWithEmojis = emojiNameToUnicode(commits)
 
-  const allowedChanges = changesWithEmojis
-    .split('\n')
-    .filter(isNotVersionEntry)
-    .filter(isNotMaintenanceEntry)
-    .sort(byEmojiPriority)
-    .map((entry) => `- ${entry}`)
-    .join('\n')
-
-  // changes with pull request links
-  return allowedChanges.replace(
-    /\(#(\d+)\)/gm,
-    (_, id) => `([#${id}](https://github.com/DataDog/browser-sdk/pull/${id}))`
-  )
-}
+  let changes = changesWithEmojis.split('\n').filter(isNotVersionEntry)
+  let internalChanges = []
+  let publicChanges = []
 
-function isNotVersionEntry(line) {
-  return !/^v\d+\.\d+\.\d+/.test(line)
+  changes.forEach((entry) => {
+    let trimmedEntry = entry.trim()
+    if (PUBLIC_EMOJI_PRIORITY.some((emoji) => trimmedEntry.startsWith(emoji))) {
+      publicChanges.push(entry)
+    } else {
+      internalChanges.push(entry)
+    }
+  })
+
+  const sortAndFormat = (entries, priority) =>
+    entries.sort((a, b) => sortByEmojiPriority(a, b, priority)).map((entry) => `- ${entry}`)
+  internalChanges = sortAndFormat(internalChanges, INTERNAL_EMOJI_PRIORITY)
+  publicChanges = sortAndFormat(publicChanges, PUBLIC_EMOJI_PRIORITY)
+
+  return `
+**Public Changes:**
+
+${publicChanges.join('\n')}
+
+**Internal Changes:**
+
+${internalChanges.join('\n')}
+`.replace(/\(#(\d+)\)/gm, (_, id) => `([#${id}](https://github.com/DataDog/browser-sdk/pull/${id}))`)
```

Comment:
**❓ question:** ‏this is nice, but why is it done only for internal changes?

Reply:
Thanks ! What do you mean only for internal changes ? it's replacing for internal and public I guess

Reply:
Ah sorry I misread! Looks fine then.